### PR TITLE
Monitoring - Add instrumentation

### PR DESCRIPTION
**Pull request by bttf**
This PR adds instrumentation for our BG jobs so that we can monitor:

- \# of currently running jobs for each queue
- the execution time for each job

Changes:
- Switch from default open telemetry startup script to our own custom `tracing.ts` script
  - Largely mimics the default script with a few added tweaks for our use case
- Add `trackJob` utility
- Apply `trackJob` utility to our simpler BG jobs as a first step to allow for evaluation before migrating all jobs.

## Sanity tests
Tested kicking off BG jobs `expireOldQueries` and `createAutoGeneratedMetrics` with success

expireOldQueries
Screenshot 2024-02-13 at 10 09 33 AM

createAutoGeneratedMetrics
Screenshot 2024-02-13 at 10 16 08 AM

### CODE REVIEW

**Comment by bttf** [comment]
@jdorn This has been tested on my local and works fine in a scenario where OTEL env vars are not defined. I think we'd still want to test the agenda jobs, however my stats engine / python setup is still broken.

**Comment by bttf** [comment]
This has fallen behind but we still intend to merge this. The team won't be available this week but plan is to get this up-to-date and merged sometime next week. @itsgrimetime Were you able to diagnose why histogram data wasn't transmitted? I took a look but nothing obvious stood out. Not sure if it may require some add'l config on the Datadog side.

File: packages/back-end/src/services/otel.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      histogram?.record(new Date().getTime() - startTime);
+    } catch (e) {
+      logger.error(`error reporting count metric for job: ${jobName} - ${e}`);
+    }
+
+    return res;
+  } catch (e) {
+    try {
+      counter?.add(-1);
+      histogram?.record(new Date().getTime() - startTime);
```
Comment (jdorn) [inline packages/back-end/src/services/otel.ts:51]: Should we also have a counter for job failures?  I can see that being useful

File: packages/back-end/src/tracing.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import { OTLPTraceExporter } from "@opentelemetry/exporter-trace-otlp-proto";
+import { OTLPMetricExporter } from "@opentelemetry/exporter-metrics-otlp-proto";
+import * as opentelemetry from "@opentelemetry/sdk-node";
+import { getNodeAutoInstrumentations } from "@opentelemetry/auto-instrumentations-node";
+import { Resource } from "@opentelemetry/resources";
+import { SemanticResourceAttributes } from "@opentelemetry/semantic-conventions";
+import { PeriodicExportingMetricReader } from "@opentelemetry/sdk-metrics";
+
+const resource = new Resource({
+  [SemanticResourceAttributes.SERVICE_NAME]: "adnans-growthbook",
```
Comment (bttf) [inline packages/back-end/src/tracing.ts:11]:
```suggestion
  [SemanticResourceAttributes.SERVICE_NAME]: "growthbook",
```

Comment (jdorn) [inline packages/back-end/src/tracing.ts:11]: Don't hardcode this

File: packages/back-end/package.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
     "dev": "yarn build:clean && concurrently \"yarn watch-compile\" \"yarn watch-dev | yarn pino-pretty\"",
     "build": "yarn build:clean && yarn build:typescript && yarn build:emails",
     "start": "node dist/server.js",
-    "start:with-tracing": "node --require @opentelemetry/auto-instrumentations-node/register dist/server.js",
+    "start:with-tracing": "node --require dist/tracing.js dist/server.js",
```
Comment (jdorn) [inline packages/back-end/package.json:16]: Is there a good reason we need to change this?  If not, i would revert `package.json` and `yarn.lock`.  Will make the PR much less risky

**Comment by bttf** [comment]
Going to start a new one

**Pull request closed without merging**